### PR TITLE
Kurt/tests

### PR DESCRIPTION
**Pull request by Degiorgio**

### CODE REVIEW

File: src/java_bytecode/java_bytecode_convert_method.cpp [hunk 1]
```diff
@@ -992,9 +992,19 @@ exprt java_bytecode_convert_methodt::get_or_create_clinit_wrapper(
   symbolt bool_local;
   if(findit==symbol_table.symbols.end())
   {
-    bool_local=add_new_symbol(symbol_table,
-      "__CPROVER_was_ready","__CPROVER_was_ready",
-      bool_typet(), false_exprt(), false, false);
+    const irep_idt& name="__CPROVER_was_ready";
```
Comment (cesaro) [inline src/java_bytecode/java_bytecode_convert_method.cpp:995]: Kurt things there is abug here.

File: src/Makefile [hunk 2]
```diff
@@ -38,6 +38,9 @@ goto-diff.dir: languages goto-programs.dir pointer-analysis.dir \
 
 goto-cc.dir: languages pointer-analysis.dir goto-programs.dir linking.dir
 
+java-models:
+	$(MAKE) $(MAKEARGS) -C "java_bytecode/library"
```
Comment (cesaro) [inline src/Makefile:42]: That space should be a tab!

File: regression/cbmc-java-concurrency/anonymous_thread_runnable/A.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import java.lang.Thread;
+import org.cprover.CProver;
+
+public class A
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/anonymous_thread_runnable/A.java:4]: Add 1 line here saying what you want to test

File: regression/cbmc-java-concurrency/anonymous_thread_runnable/test.desc [hunk 4]
```diff
@@ -0,0 +1,7 @@
+CORE
+A.class
+--function "A.me:()V" --lazy-methods --lazy-methods --classpath .:../../../src/java_bytecode/library/core-models.jar:.
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/anonymous_thread_runnable/test.desc]: --lazy-methods twice!

File: regression/cbmc-java-concurrency/extending_thread/A.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  public void me()
+  {
+    Thread tmp = new Thread(this);
+    tmp.start();
+    setX();
+  }
+
+  public void setX()
+  {
+    x=10;
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/extending_thread/A.java]: add spaces here

File: regression/cbmc-java-concurrency/extending_thread/A.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    x = 44;
+    int local_x = x;
+    assert(local_x == 44);
+  }
+
+  public void me()
+  {
+    Thread tmp = new Thread(this);
+    tmp.start();
+    x=10;
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/extending_thread/A.java]: spaces here

File: regression/cbmc-java-concurrency/extending_thread/A.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+  // expected verfication failed
+  public void me2()
+  {
+    B b = new B();
+    b.me();
+  }
+
+  // expected verfication success.
+  public void me()
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/extending_thread/A.java:33]: Reorder

File: regression/cbmc-java-concurrency/extending_thread/test.desc [hunk 8]
```diff
@@ -0,0 +1,7 @@
+CORE
+A.class
+--function "A.me:()V" --lazy-methods --lazy-methods --classpath .:../../../src/java_bytecode/library/core-models.jar:.
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/extending_thread/test.desc]: duble lazy methods

File: regression/cbmc-java-concurrency/implementing_runnable/test4.desc [hunk 9]
```diff
@@ -0,0 +1,7 @@
+CORE
+A.class
+--function "A.me3:()V" --lazy-methods --classpath .:../../../src/java_bytecode/library/core-models.jar:.
+
+^EXIT=0$
+^SIGNAL=0$
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/implementing_runnable/test4.desc:6]: Add java file in this folder.

File: regression/cbmc-java-concurrency/several_threads/A.java [hunk 10]
```diff
@@ -0,0 +1,97 @@
+import java.lang.Thread;
+import org.cprover.CProver;
+
+public class A
+{
+
+  // expected verification success
+  public void me()
+  {
+    B b= new B();
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/several_threads/A.java]: space

File: regression/cbmc-java-concurrency/several_threads/A.java [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+    int local_x = x;
+    assert(local_x == 44 || local_x == 10);
+  }
+  public void setX()
+  {
+    x=10;
+  }
+}
+
+
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/several_threads/A.java]: Remove lines here

File: regression/cbmc-java-concurrency/several_threads/A.java [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  // expected verification failure
+  public void me2()
+  {
+    C c= new C();
+    c.me();
+  }
+
+  // Create 32 Threads, no shared variables
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/several_threads/A.java]: - modify test to have 2 threads
- modify comment accordinly
- write FIXME saying that this shoud pass with 32 threads in reasonable time

File: regression/cbmc-java-concurrency/several_threads/A.java [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+    x=10;
+  }
+}
+
+
+
+
+class B implements Runnable
+{
+  int x = 0;
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/several_threads/A.java]: Add comment:
```java
// FIXME: the assertion below does not always hold as per the Java spec (because x is not volatile), but cbmc currently doesn't reason about volatile variables
```

File: regression/cbmc-java-concurrency/several_threads/A.java [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+  }
+
+  // verfication success
+  public void me()
+  {
+    Thread tmp = new Thread(this);
+    tmp.start();
+    tmp.start();
+    tmp.start();
+    x=10;
```
Comment (cesaro) [inline regression/cbmc-java-concurrency/several_threads/A.java]: spacing

File: src/java_bytecode/CMakeLists.txt [hunk 15]
```diff
@@ -1,7 +1,6 @@
 get_filename_component(JAVA_CORE_LIBRARY_PATH "library" REALPATH BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 get_filename_component(CORE_MODELS_JAR "library/core-models.jar" REALPATH BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 file(GLOB_RECURSE jars "*.jar")
```
Comment (cesaro) [inline src/java_bytecode/CMakeLists.txt]: Make this a different commit.

File: regression/cbmc-java-concurrency/anonymous_thread_runnable/test.desc (hunk 4 above)
Comment (Degiorgio) [reply regression/cbmc-java-concurrency/anonymous_thread_runnable/test.desc]: done
